### PR TITLE
TEST/UCP/GDR_COPY: Add gdr_copy test

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: test/gtest/ucp/ucp_test.h [hunk 1]
```diff
@@ -470,6 +470,8 @@ std::vector<ucp_test_param> enum_test_params(const std::string& tls)
     UCP_INSTANTIATE_TEST_CASE_TLS_GPU_AWARE(_test_case, ugni, \
                                             "ugni") \
     UCP_INSTANTIATE_TEST_CASE_TLS_GPU_AWARE(_test_case, tcp, \
-                                            "tcp")
+                                            "tcp") \
+    UCP_INSTANTIATE_TEST_CASE_TLS_GPU_AWARE(_test_case, shm_gdr, \
```
Comment (yosefe) [inline test/gtest/ucp/ucp_test.h:474]: maybe add to to shm_ib tests, to not increase testing matrix and CI time?

File: src/uct/cuda/gdr_copy/gdr_copy_md.c [hunk 2]
```diff
@@ -58,6 +51,7 @@ uct_gdr_copy_md_query(uct_md_h md, uct_md_attr_v2_t *md_attr)
     md_attr->max_reg                = ULONG_MAX;
     md_attr->rkey_packed_size       = sizeof(uct_gdr_copy_key_t);
     md_attr->reg_cost               = UCS_LINEAR_FUNC_ZERO;
+    md_attr->reg_alignment          = UCT_GDR_COPY_MD_RCACHE_DEFAULT_ALIGN;
```
[the remark this answers is not in the document]
Comment (Artemy-Mellanox) [reply src/uct/cuda/gdr_copy/gdr_copy_md.c:54]: maybe use GPU_PAGE_SIZE from gdrapi.h?

[the remark this answers is not in the document]
Comment (yosefe) [reply src/uct/cuda/gdr_copy/gdr_copy_md.c:54]: > maybe use GPU_PAGE_SIZE from gdrapi.h?

Sounds good.

**Review by yosefe — DISMISSED** [review]

**Comment by yosefe** [comment]
@Artemy-Mellanox can you pls rebase?

**Review by yosefe — APPROVED** [review]

**Pull request merged**